src/bootstrap/src/core/build_steps/tool.rs

```diff
@@ -1096,6 +1096,50 @@ tool_extended!((self, builder),
     Rustfmt, "src/tools/rustfmt", "rustfmt", stable=true, add_bins_to_sysroot = ["rustfmt", "cargo-fmt"];
 );
 
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct TestFloatParse {
+    pub path: PathBuf,
```

Comment:
This isn't used in `tool` build, it's more for running tests in `test` module.

Reply:
ok will move the whole thing to `test`

Reply:
I didn't mean that, I was pointing the `path` part.

Reply:
ok got it